### PR TITLE
Various refactor

### PR DESCRIPTION
**Pull request by Janmm14**
- Started to migrate gomint-server to fluid accessors
- Qualified instance field access with `this` always
- Fix ItemEnchantEvent levelRequirement was not set in constructor
- Events got equals, hashCode removed as its not needed
- Events got toString removed because of bad implementation (ignoring fields of super classes)
- Various javadoc improvements and fixes
- Event class hierarchy improvements with interfaces
- Code formatting in edited classes and a couple others, we're still slowly migrating away from spacy layout.
- .gitignore: Add tileentity registry (as it's a generated file)

Likely breaks API due to the event changes, at least on bytecode level.

Wolrd multithreading PR #727 depends on this.

### CODE REVIEW

**Comment by Janmm14** [comment]
Should PlayerEvent extend EntityEvent?

File: gomint-api/src/main/java/io/gomint/event/interfaces/PlayerEvent.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import io.gomint.world.World;
+
+/**
+ * Marks an event that takes place in relation to a player
+ *
+ * @author Janmm14
+ * @version 2.0
+ * @stability 1
+ */
+public interface PlayerEvent extends WorldEvent {
```
Comment (geNAZt) [inline gomint-api/src/main/java/io/gomint/event/interfaces/PlayerEvent.java]: This should extend entity event

File: gomint-server/src/main/java/io/gomint/server/entity/projectile/EntityExpBottle.java [hunk 2]
```diff
@@ -60,8 +60,8 @@ protected void applyCustomProperties() {
         super.applyCustomProperties();
 
         // Gravity
-        GRAVITY = 0.07f;
-        DRAG = 0.01f;
+        this.GRAVITY = 0.07f;
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/entity/projectile/EntityExpBottle.java]: We should lowercase those in a refactor too

**Review by geNAZt — CHANGES_REQUESTED** [review]

**Comment by Janmm14** [comment]
Event debug logging regressed due to removal of event toString methods, but I think that's acceptable.

In the future this can be readded, maybe it'd be ok to use `ReflectionToStringBuilder` or some similar thing as there's no toString generator which includes with fields from super classes via their fluent accessors.

**Review by geNAZt — APPROVED** [review]

**Pull request merged**